cdap-master/src/main/java/co/cask/cdap/master/startup/ConfigurationCheck.java

```diff
@@ -191,8 +191,8 @@ private boolean isValidKafkaTopic(String key) {
 
   private boolean isValidPartitionKey(String key) {
     try {
-      LogPartitionType.valueOf(key);
-    } catch (IllegalArgumentException e) {
+      LogPartitionType.valueOf(key.toUpperCase());
+    } catch (IllegalArgumentException | NullPointerException e) {
```

Comment:
The issue was that the common jar was out of date (someone else had used my cluster and manually copied over some jars) and its cdap-default.xml did not contain the property. The issue here was that we call toUpper outside of the try-catch, so the property name is not added to the problematic keys and you have no idea what happened. 

I think we should write our code in ways that does not assume that everything is always consistent. It is ok to fail when there is an inconsistency. But it should fail with a sensible error message. Whereas this startup test failed without logging the exception, and and with a message of null. That made it impossible to diagnose - I had to add extra logging to find out what the offending key was. 

The fix is to move the toUpper() into the try-catch, and also to print the exception name in the log, because any time we have an NPE (and maybe for other issues, too) the message is null.